imap_processing/spice/geometry.py

```diff
@@ -87,3 +90,67 @@ def imap_state(
         SpiceBody.IMAP.name, et, ref_frame.name, "NONE", observer.name
     )
     return np.asarray(state)
+
+
+def get_spin_data() -> pd.DataFrame:
+    """
+    Read spin file using environment variable and return spin data.
+
+    SPIN_DATA_FILEPATH environment variable would be a fixed value.
+    It could be s3 filepath that can be used to download the data
+    through API or it could be path EFS or Batch volume mount path.
+
+    Spin data should contains the following fields:
+        (
+            spin_number,
+            spin_start_sec,
+            spin_start_subsec,
+            spin_period_sec,
+            spin_period_valid,
+            spin_phase_valid,
+            spin_period_source,
+            thruster_firing
+        )
+
+    Returns
+    -------
+    spin_data : pandas.DataFrame
+        Spin data.
+    """
+    spin_data_filepath = os.getenv("SPIN_DATA_FILEPATH")
+    if spin_data_filepath is not None:
+        path_to_spin_file = Path(spin_data_filepath)
+    else:
+        # Handle the case where the environment variable is not set
+        raise ValueError("SPIN_DATA_FILEPATH environment variable is not set.")
+
+    return pd.read_csv(path_to_spin_file)
+
+
+def get_spacecraft_spin_phase(
+    query_met_times: Union[float, np.ndarray[float]],
+) -> Union[float, np.ndarray]:
+    """
+    Get the spacecraft spin phase for the input query times.
+
+    Formula to calculate spin phase:
+        spin_phase = (
+            query_met_times - (spin_start_seconds + spin_start_subseconds)
+        ) / spin_period_sec
+
+    Parameters
+    ----------
+    query_met_times : float or np.ndarray[float]
+        Query times in Mission Elapsed Time (MET).
+
+    Returns
+    -------
+    spin_phase : float or np.ndarray
+        Spin phase for the input query times.
+    """
+    # TODO: Write code to calculate spin phase for the input query times
+    if isinstance(query_met_times, float):
+        query_met_times = np.array([query_met_times])
+    # TODO: call get_spin_data function to get spin data for the input
+    # query times. And finish remaining implementation in upcoming PR.
+    return np.zeros_like(query_met_times)
```

Comment:
My last nit pick is that the type hint seems to indicate that if `query_met_times` is a float, then this function should return a float, not an array.

Reply:
Where do you see this type hint? I am trying to find it but don't see it on my end.

Reply:
got it. I was not paying close attention to this function. I should have return float for float input value instead of np.array.

Reply:
Sorry, I wasn't very clear. I think that if `get_spacecraft_spin_phase` gets a float for `query_met_times` that it is meant to return a float based on the function defined typing and the Returns doc string.